Support Storybook for the common package

sgratch opened the pull request:
Fixes: #414 

Support the Storybook (documentation and examples) for the components of the common package.

### Few Screenshots:
#### LoadingDots component
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/87164b0c-177c-4828-af77-d9cbb8e35c78)
#### SwitchFilter component
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/93efb136-615b-4847-b1b0-261bf51fa1f1)
#### FilterGroup component
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/12655560-106b-4b49-9d15-49bad811ea36)
#### TableView component
![image](https://github.com/kubev2v/forklift-console-plugin/assets/18169498/569eb0be-785c-4fd7-9c18-4ce0bc095338)


### For displaying the Storybook UI:
1.  please run the following command under the `common` package:
`npm run storybook`

2. Open a browser to `http://localhost:6006/`

### The Following components are currently documented by Storybook:
1. Filter - EnumFilter, FreetextFilter, GroupedEnumFilter, SwitchFilter
2. FilterGroup - AttributeValueFilter, FilterGroup
3. Page - BaseState, NoResultsMatchFilter
4. TableView - DefaultHeader, ManageColumnsModal, ManageColumnsToolbarItem, TableView, 
5. ExternalLink
6. LoadingDots
7. RedHatProgressionIcon

### The Following components are not documented by Storybook, due to technical reasons:
1. ActionServiceDropdown - since based on dynamic-plugin-sdk
2. QueryClientHoc - not a React/UI component
3. Page - loadUserSettings, usePagination, useFields - not React/UI component

sgratch commented:
Note: this depends on #513. Please avoid merging till #513 is merged.

yaacov reviewed (approved):
LGTM

yaacov commented on packages/common/package.json:
```diff
@@ -32,6 +34,26 @@
+    "sass": "^1.63.3",
+    "storybook": "^7.0.14"
+  },
+  "dependencies": {
```
@sgratch we should not depend on story book in our release code

can you move this so we will not deploy the storybook code to users ?

sgratch replied:
Done.

sgratch commented:
@yaacov the linter removed the spaces I added for newlines, so I replaced them with a line break for a better documentation alignment in storybook. 
Now it's ready for review.
